### DISCUSSION TITLE
IPI Ovirt

### DISCUSSION THREAD

**Discussion by mathgonzlez** [okd-project/okd#1801] [General]
**Describe the bug**
After installing OKD (version 4.13.0-0.okd-2023-06-04-080300) using the IPI method within RHV 4.4SP1, the cluster was successfully installed.
I conducted a test by gracefully shutting down the cluster as per the documentation: [Graceful Cluster Shutdown](https://docs.openshift.com/container-platform/4.13/backup_and_restore/graceful-cluster-shutdown.html), and everything worked without any issues. 
However, when attempting to shut down the nodes forcibly, by forcefully stopping the VMs to simulate an unexpected failure, the cluster could not recover. 
While one of the workers could successfully obtain both an IP and the VIPs for Ingress and API, the other nodes displayed the following messages:

**Version**
OKD version: 4.13.0-0.okd-2023-06-04-080300
Install method: IPI
Cloud platform: RHV 4.4SP1

**How reproducible**
Always

**Log bundle**
I cant share must-gather because my cluster is unhealty

Thanks

**Comment by vrutkovs** [comment]
>the cluster could not recover.

Do masters boot?

>While one of the workers could successfully obtain both an IP and the VIPs for Ingress and API,

Why would a worker get an IP for API?

**Comment by mathgonzlez** [comment]
Sorry, it was a mistake, I meant the control planes. If the nodes boot.
![image](https://github.com/okd-project/okd/assets/57609433/68603fac-74c5-439e-ad10-6c86a70bfa65)
![image](https://github.com/okd-project/okd/assets/57609433/2ce48ea6-c33a-4140-9c28-b320e62e0dee)


EDIT: after a couple of minutes this is my IP asignation:
![image](https://github.com/okd-project/okd/assets/57609433/75bf8017-b5ad-42bf-ac40-49b51a860711)

I access to master-2 and see the afterburn-service in failed state:


> × afterburn-sshkeys@core.service - Afterburn (SSH Keys)
     Loaded: loaded (/usr/lib/systemd/system/afterburn-sshkeys@.service; enabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: failed (Result: exit-code) since Mon 2023-11-13 20:48:06 UTC; 21min ago
    Process: 852 ExecStart=/usr/bin/afterburn ${AFTERBURN_OPT_PROVIDER} --ssh-keys=core (code=exited, status=1/FAILURE)
   Main PID: 852 (code=exited, status=1/FAILURE)
        CPU: 106ms
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]:     0: writing ssh keys
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]:     1: maximum number of retries (10) reached
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]:     2: failed to fetch
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]:     3: error sending request for url (http://169.254.169.254/latest/meta-data/public-keys): error trying to connect: tcp connect error: Network is unreachable (os error 101)
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]:     4: error trying to connect: tcp connect error: Network is unreachable (os error 101)
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]:     5: tcp connect error: Network is unreachable (os error 101)
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]:     6: Network is unreachable (os error 101)
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 systemd[1]: afterburn-sshkeys@core.service: Main process exited, code=exited, status=1/FAILURE
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 systemd[1]: afterburn-sshkeys@core.service: Failed with result 'exit-code'.
Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 systemd[1]: Failed to start afterburn-sshkeys@core.service - Afterburn (SSH Keys).

**Comment by vrutkovs** [comment]
>Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]: 3: error sending request for url (http://169.254.169.254/latest/meta-data/public-keys): error trying to connect: tcp connect error: Network is unreachable (os error 101)

afterburn is unable to reach metadata service.

**Comment by mathgonzlez** [comment]
> > Nov 13 20:48:06 okd-ovirt-cdgkj-master-2 afterburn[852]: 3: error sending request for url (http://169.254.169.254/latest/meta-data/public-keys): error trying to connect: tcp connect error: Network is unreachable (os error 101)
> 
> afterburn is unable to reach metadata service.

Yes i see that, But why this only happend when i shutdown my nodes in a not gracefully?

**Reply by vrutkovs** [reply]
Not much I can help with without a must-gather

**Reply by mathgonzlez** [reply]
Could I redeploy the cluster and share a must-gather before shutdown the cluster, do you think it could help?
